src/components/views/globals/ServerLimitBar.js

```diff
@@ -15,15 +15,60 @@ limitations under the License.
 */
 
 import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import { _t } from '../../../languageHandler';
 
 export default React.createClass({
+    propTypes: {
+        // 'hard' if the logged in user has been locked out, 'soft' if they haven't
+        kind: PropTypes.string,
+        adminContent: PropTypes.string,
+    },
+
+    getDefaultProps: function() {
+        return {
+            kind: 'hard',
+        }
+    },
+
     render: function() {
-        const toolbarClasses = "mx_MatrixToolbar mx_MatrixToolbar_error";
+        const toolbarClasses = {
+            'mx_MatrixToolbar': true,
+        };
+        let content;
+
+        const translateLink = (sub) => {
+            if (this.props.adminContent) {
+                return <a href={this.props.adminContent}>{sub}</a>;
+            } else {
+                return sub;
+            }
+        }
+
+        if (this.props.kind === 'hard') {
+            toolbarClasses['mx_MatrixToolbar_error'] = true;
+            content = _t(
+                "This homeserver has hit its Monthly Active User limit. Please <a>contact your service administrator</a> to continue using the service.",
```

Comment:
This would be shown when (initial) sync has failed I think, but wouldn't it be very confusing to just show a banner, but still the complete UI, but you can't interact with the server in any meaningful way? (or is that not what M_MAU_LIMIT_EXCEEDED causes?). Maybe I'm missing something, but cancelling login and showing an error message on the login page would be clearer if this is the case.

Reply:
I think the idea is that you can still see the messages you've already synced, although it's tied quite heavily into the fact that causing a logout causes the client to delete its e2e keys: I think this may be the intention in the long term once we work out how to do it.